### PR TITLE
[build] Use more dependencies from mapbox-base

### PR DESCRIPTION
**Pull request by pozdnyakov**
This PR updates `mapbox-base` version and uses the following dependencies from there:
- [x] args
- [x] filesystem
- [x] kdbush.hpp
- [x] rapidjson
- [x] geojson.hpp
- [x] jni.hpp
- [x] pixelmatch-cpp
- [x] supercluster.hpp

### CODE REVIEW

File: cmake/vendor.cmake [hunk 1]
```diff
@@ -9,29 +9,32 @@ include(${CMAKE_SOURCE_DIR}/vendor/cheap-ruler-cpp.cmake)
...
 include(${CMAKE_SOURCE_DIR}/vendor/wagyu.cmake)
-include(${CMAKE_SOURCE_DIR}/vendor/args.cmake)
 
 if(NOT TARGET mapbox-base)
-    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/vendor/mapbox-base ${CMAKE_BINARY_DIR}/.build-mapbox-gl-native-mapbox-base)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/vendor/mapbox-base ${CMAKE_BINARY_DIR}/.build-mapbox-gl-native-mapbox-base EXCLUDE_FROM_ALL)
 endif()
 
+target_compile_definitions(mapbox-base-extras-rapidjson INTERFACE
+    RAPIDJSON_HAS_STDSTRING=1
```
Comment (tmpsantos) [inline cmake/vendor.cmake]: This should come from mapbox-base

File: cmake/vendor.cmake [hunk 2]
```diff
@@ -9,29 +9,32 @@ include(${CMAKE_SOURCE_DIR}/vendor/cheap-ruler-cpp.cmake)
...
+)
+
+if(WIN32)
+    target_compile_definitions(mapbox-base-extras-rapidjson INTERFACE
+        RAPIDJSON_HAS_CXX11_RVALUE_REFS
```
Comment (tmpsantos) [inline cmake/vendor.cmake]: Ditto.

File: cmake/core.cmake [hunk 3]
```diff
@@ -28,12 +26,12 @@ endif()
...
 target_link_libraries(mbgl-core PUBLIC
     boost
+    mapbox-base-extras
     mapbox-base
-    geojson.hpp
     ${ICU_LIBRARY}
     polylabel
     protozero
-    rapidjson
+    Mapbox::Base::Extras::rapidjson
```
Comment (alexshalamov) [inline cmake/core.cmake]: Just curious why rapidjson is in extras? It is mandatory component for code, while, e.g.,  pixelmatch is not.

File: cmake/vendor.cmake [hunk 4]
```diff
@@ -9,29 +9,32 @@ include(${CMAKE_SOURCE_DIR}/vendor/cheap-ruler-cpp.cmake)
...
 include(${CMAKE_SOURCE_DIR}/vendor/vector-tile.cmake)
 include(${CMAKE_SOURCE_DIR}/vendor/wagyu.cmake)
-include(${CMAKE_SOURCE_DIR}/vendor/args.cmake)
 
 if(NOT TARGET mapbox-base)
-    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/vendor/mapbox-base ${CMAKE_BINARY_DIR}/.build-mapbox-gl-native-mapbox-base)
+    add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/vendor/mapbox-base ${CMAKE_BINARY_DIR}/.build-mapbox-gl-native-mapbox-base EXCLUDE_FROM_ALL)
 endif()
 
+target_compile_definitions(mapbox-base-extras-rapidjson INTERFACE
```
Comment (alexshalamov) [inline cmake/vendor.cmake]: could this be part of base? Same for the win32 specific compile def.

File: cmake/core.cmake (hunk 3 above)
Comment (pozdnyakov) [reply cmake/core.cmake]: AFAIK `extras` in mapbox-base means just 3rd parties cc @brunoabinader

**Pull request merged**